src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8513,44 +8519,198 @@ protected MethodGroupResolution ResolveExtension(
+                    methodResult.Free(keepArguments: true);
+                    propertyResult?.Free();
+                    result = makeErrorResult(left.Type, memberName, arity, lookupResult, expression, diagnostics);
+                    return true;
+                }
+
+                // If the search in the current scope resulted in any applicable property (regardless of whether a best
+                // applicable property could be determined) then our search is complete.
+                Debug.Assert(propertyResult?.HasAnyApplicableMember == true);
+                methodResult.Free(keepArguments: true);
+                if (propertyResult.Succeeded && propertyResult.BestResult.Member is { } bestProperty)
+                {
+                    // property wins
+                    propertyResult.Free();
+                    result = new MethodGroupResolution(bestProperty, LookupResultKind.Viable, diagnostics.ToReadOnly());
+                    return true;
+                }
+
+                // ambiguous between multiple applicable properties
+                propertyResult.Free();
+                // PROTOTYPE consider using the property overload resolution result in the result to improve reported diagnostics
+                result = makeErrorResult(left.Type, memberName, arity, lookupResult, expression, diagnostics);
+                return true;
+            }
+
+            static MethodGroupResolution resolveMethods(
+                SyntaxNode expression,
+                BoundExpression left,
+                ImmutableArray<TypeWithAnnotations> typeArgumentsWithAnnotations,
+                TypeSymbol? returnType,
+                RefKind returnRefKind,
+                LookupResult lookupResult,
+                AnalyzedArguments? analyzedArguments,
+                ref AnalyzedArguments? actualMethodArguments,
+                OverloadResolution.Options options,
+                in CallingConventionInfo callingConvention,
+                Binder binder,
+                BindingDiagnosticBag diagnostics)
+            {
+                ArrayBuilder<Symbol>? methods = null;
+                foreach (var member in lookupResult.Symbols)
+                {
+                    if (member is MethodSymbol method)
+                    {
+                        methods ??= ArrayBuilder<Symbol>.GetInstance();
+                        methods.Add(method);
+                    }
+                }
+
+                if (methods is null)
+                {
+                    return default;
                 }
 
                 var methodGroup = MethodGroup.GetInstance();
-                methodGroup.PopulateWithExtensionMethods(left, members, typeArgumentsWithAnnotations, resultKind: lookupResult.Kind);
-                members.Free();
+                methodGroup.PopulateWithExtensionMethods(left, methods, typeArgumentsWithAnnotations, resultKind: lookupResult.Kind);
```

Comment:
>PopulateWithExtensionMethods [](http://example.com/codeflow?start=28&length=28)

Consider if it would make sense to adjust ```PopulateWithExtensionMethods``` to perform filtering for methods